RecoTauTag/RecoTau/plugins/PFRecoTauChargedHadronFromTrackPlugin.cc

```diff
@@ -143,13 +143,14 @@ PFRecoTauChargedHadronFromTrackPlugin::return_type PFRecoTauChargedHadronFromTra
   evt.getByToken(Tracks_token, tracks);
 
   qcuts_->setPV(vertexAssociator_.associatedVertex(jet));
-
+  float jEta=jet.eta();
+  float jPhi=jet.phi();
   size_t numTracks = tracks->size();
   for ( size_t iTrack = 0; iTrack < numTracks; ++iTrack ) {
     reco::TrackRef track(tracks, iTrack);
 
     // consider tracks in vicinity of tau-jet candidate only
-    double dR = deltaR(track->eta(), track->phi(), jet.eta(), jet.phi());
+    double dR = deltaR(track->eta(), track->phi(), jEta,jPhi);
     double dRmatch = dRcone_;
     if ( dRconeLimitedToJetArea_ ) {
```

Comment:
interesting…

deltaR is everywhere…

Is there a (twiki) summary of deltaR* tricks? I’m pretty lost myself just looking at the code

> On Mar 4, 2015, at 1:34 PM, Vincenzo Innocente <notifications@github.com> wrote:
> 
> In RecoTauTag/RecoTau/plugins/PFRecoTauChargedHadronFromTrackPlugin.cc:
> 
> >    size_t numTracks = tracks->size();
> >    for ( size_t iTrack = 0; iTrack < numTracks; ++iTrack ) {
> >      reco::TrackRef track(tracks, iTrack);
> >  
> >      // consider tracks in vicinity of tau-jet candidate only
> > -    double dR = deltaR(track->eta(), track->phi(), jet.eta(), jet.phi());
> > +    double dR = deltaR(track->eta(), track->phi(), jEta,jPhi);
> >      double dRmatch = dRcone_;
> >      if ( dRconeLimitedToJetArea_ ) {
> 
> while factorizing also this part can be moved outside the loop...
> and (deltaR moved to deltaR2)
> (we should ban deltaR)
> 
> —
> Reply to this email directly or view it on GitHub.
>